`mephisto-task` docs update

pringshia opened the pull request:
Docs for `mephisto-task`

JackUrb commented on packages/mephisto-task/README.md:
```diff
@@ -46,56 +58,62 @@ function MyApp() {
 ### `initialTaskData`
 
-The initial data to populate your task with. The crowdworker will use this data to perform their action.
+The initial data to populate your task with. The worker will use this data to perform their action.
```
Useful to note where this comes from on the backend - it's the `InitializationData` object that the TaskRunner returns in `get_data_for_assignment`, which comes from the set of `InitializationData` objects returned by the blueprint's `get_initialization_data` method.

JackUrb commented on packages/mephisto-task/README.md:
```diff
@@ -46,56 +58,62 @@ function MyApp() {
 
 ### `taskConfig`
-A config object passed from the backend.
+An arbitrary task-specific config object passed from the backend.
```
Useful to note where this comes from, it's the `get_frontend_args` method of the blueprint.

JackUrb reviewed (commented):
Some small comments - this is definitely getting to a place that I think is more understandable, but I think it may still be hard to get a new user to try using `mephisto-task` hooks directly. Unsure what the right thing to do for that is yet though.

pringshia commented on packages/mephisto-task/README.md:
```diff
@@ -46,56 +58,62 @@ function MyApp() {
+
 ### `assignmentId`
 
+When a worker accepts a task, an `assignmentId` is created to represent the pairing between the worker and the task.
+
```
You're right. Mixed it up.

pringshia commented:
[**View changes since last update**](https://github.com/facebookresearch/Mephisto/pull/274/files/4400e949971f359bdba6e621bff289ee32eb91c3..090b8bafd246cccc979196882c3424db204515e3)

JackUrb reviewed (approved):
These docs really help clarify how the system works. ❤️  I suspect this will allow new users to ramp into `mephisto-task` more quickly.
